### PR TITLE
[FLOSS T2] tutorial: update array tutorial

### LINKED ISSUE #4250: [FLOSS T2] update array tutorial
for natural sorting (_ are not needed anymore)

https://www.libelektra.org/tutorials/arrays

see also #4249

### PR DESCRIPTION
**Pull request by Bujuhu**
fixes #4250

Co-authored-by: Dynamichost96 <60710121+Dynamichost96@users.noreply.github.com>

We worked on this issue entirely in pair

The automatic validation of the tutorial is executed in jenkins

## Basics

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: doc/tutorials/arrays.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ kdb get user:/tests/parent/daughter
 
 ## Array Keys
 
-Since Elektra sorts keys alphabetically we can use the key-value pair structure described above to store sequences of values (aka. arrays).
+Since Elektra keys are sorted, we can use the key-value pair structure described above to store sequences of values (aka. arrays).
```
Comment (0x6178656c) [inline doc/tutorials/arrays.md:46]: Why did you remove "alphabetically"?

File: doc/tutorials/arrays.md [hunk 2]
```diff
@@ -72,14 +72,14 @@ kdb set user:/tests/sequence/#3 'Fourth Element'
 
 . As you can see above arrays can contain “empty fields”: The key `user:/tests/sequence/#2` is missing.
 
-For array elements with an index larger than `9` we must add **underscores** (`_`) to the basename, so we do not destroy the alphabetic order of the array. For example, to add an eleventh element to our array we use the following command:
+While regular keys are sorted alphabetically, array elements are sorted by natural ordering:
```
Comment (0x6178656c) [inline doc/tutorials/arrays.md:75]: I find this sentence hard to understand. How exactly are these array elements sorted by natural ordering? Does this affect the user in any way? The output of `kdb` is always alphabetical, no?

**Review by 0x6178656c — CHANGES_REQUESTED** [review]
Very important update to the arrays tutorial which presents updated examples that highlight the improved interface.

One thing which I am unhappy with is the usage of the term "natural order" (I think it was in the tutorial before too).
What exactly makes an ordering natural? If you insist on using this nomenclature then it should be defined somewhere.
I would suggest using "numeric order" instead of "natural order".

I think the failing checks should be resolved by rebasing off the current master.

**Comment by flo91** [comment]
Great update of the tutorial!
Rebasing with current master should fix the `missing man page` error.

File: doc/tutorials/arrays.md (hunk 2 above)
Comment (Bujuhu) [reply doc/tutorials/arrays.md:75]: The output stays the same, but for the input, Electra will automatically prepend the `_` to keep the natural ordering of the array equal to alphabetical ordering.

Comment (Bujuhu) [reply doc/tutorials/arrays.md:75]: I just realized, that my comment is probably a better explanation of what happens here than, what I've put in the documentation. Will update accordingly.

**Comment by Bujuhu** [comment]
jenkins build libelektra please

**Comment by deoknats861** [comment]
### Review

**Documentation** 
The documentation is clear and understandable. The changes made correlate with the suggestions made in https://github.com/ElektraInitiative/libelektra/issues/4250.

**Examples**
No new examples were added. The existing examples are well chosen and understandable.

**Code is conforming to our Coding Guidelines**
Yes.

**APIs are conforming to our Design Guidelines**
Yes.

**Code is consistent to our Design Decisions**
Yes.

**CI Execution of tutorials**
The latest CI run did not execute the validation of the tutorial. I will re-trigger Jenkins, as the build fail resulted from CI issues.

**Comment by deoknats861** [comment]
jenkins build libelektra please

**Comment by flo91** [comment]
jenkins build libelektra please

**Comment by Bujuhu** [comment]
> Very important update to the arrays tutorial which presents updated examples that highlight the improved interface.
> 
> One thing which I am unhappy with is the usage of the term "natural order" (I think it was in the tutorial before too). What exactly makes an ordering natural? If you insist on using this nomenclature then it should be defined somewhere. I would suggest using "numeric order" instead of "natural order".
> 
> I think the failing checks should be resolved by rebasing off the current master.

I must have missed the bit about the natural ordering. While it is technically correct, it's still a bit confusing and an unnecessary distinction to make. I've changed it now.

**Comment by flo91** [comment]
jenkins build libelektra please
